exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/References.kt

```diff
@@ -134,10 +111,19 @@ class Referrers<ParentID : Comparable<ParentID>, in Parent : Entity<ParentID>, C
  * @param cache Whether loaded reference entities should be stored in the [EntityCache].
  */
 class OptionalReferrers<ParentID : Comparable<ParentID>, in Parent : Entity<ParentID>, ChildID : Comparable<ChildID>, out Child : Entity<ChildID>, REF>(
-    val reference: Column<REF?>,
+    reference: Column<REF?>,
+    factory: EntityClass<ChildID, Child>,
+    cache: Boolean
+) : BaseReferrers<ParentID, Parent, ChildID, Child, REF?>(reference, factory, cache)
+
+abstract class BaseReferrers<ParentID : Comparable<ParentID>, in Parent : Entity<ParentID>, ChildID : Comparable<ChildID>, out Child : Entity<ChildID>, REF>(
```

Comment:
Will this switch cause any issue for the other reference classes that delegate to `OptionalReferrers`?

Also, is the plan to deprecate `OptionalReferrers` and then drop this new class after in favor of `Referrers` alone? If so, would it be better to just have duplicate code in the 2 existing classes, that way we'd only have to deprecate one class. Without having to introduce a new temporary class into the API.
Unless the plan is to push that second PR before this one is released, so `BaseReferrers` doesn't actually enter the API.

Reply:
It was actually open question what is better to do, and the variant to mark `OptionalReferrers` as deprecated looks good to me.

In order to avoid one more class in API it's not a problem to inherit `OptionalReferrers` from `Referrers` (but the last one should be `open`). Pushed such a variant, but we can discuss it again.